rclcpp/include/rclcpp/node_interfaces/node_graph_interface.hpp

```diff
@@ -299,6 +299,29 @@ class NodeGraphInterface
   virtual
   std::vector<rclcpp::TopicEndpointInfo>
   get_subscriptions_info_by_topic(const std::string & topic_name, bool no_mangle = false) const = 0;
+
+  /// Wait for a number of publishers to be available on a topic.
+  /**
+   * This function will block until the provided number of publishers are available or a timeout occurs.
+   *
+   * TODO(jacobperron): Detect this scenario and throw an exception
+   * This function should not be called concurrently with a running GraphListener.
+   * E.g. Do not call this after calling `rclcpp::GraphListener::get_graph_event()`.
+   * If this happens an exception is thrown.
```

Comment:
> Out of curiosity, I've tried to replicate this undefined behavior scenario and I didn't notice anything unexpected:

me neither.

if possible, could anyone elaborate what kind of exception we are going to get? i cannot see it precisely...

Reply:
Sorry, this PR is not complete. I imagine that we should guard against concurrent usage in the implementation and throw an exception (see the TODO in node_graph.cpp).